app/controllers/api/v1/notifications/requests_controller.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+class Api::V1::Notifications::RequestsController < Api::BaseController
+  before_action -> { doorkeeper_authorize! :read, :'read:notifications' }, only: :index
+  before_action -> { doorkeeper_authorize! :write, :'write:notifications' }, except: :index
+
+  before_action :require_user!
+  before_action :set_request, except: :index
+
+  after_action :insert_pagination_headers, only: :index
+
+  def index
+    with_read_replica do
```

Comment:
Could the read replica be set in an `around_action`?

Reply:
I'm trying to be consistent with the style proposed by @krainboltgreene originally. Be very explicit where read-replicas are used.

Reply:
Cool, makes sense. I guess if at some point that pattern becomes less one-off here and there and more widespread, we could reconsider that.

Separately, while working on a serializer performance branch, I noticed a sort of mix of styles around how the `relationships` are passed into serializers. This PR is open to standardize them - https://github.com/mastodon/mastodon/pull/29033/files - that could apply here as well (leave the `@requests` assignment in the action wrapped in with_read_replica, but move the relationships presenter wrapping to a private method). All of that is in service of reducing diff size on eventual serializer perf PR ... so not critical, but will make it easier.